src/rwlock.rs

```diff
@@ -1,20 +1,14 @@
 use std::cell::UnsafeCell;
 use std::fmt;
 use std::future::Future;
-use std::mem;
+use std::mem::ManuallyDrop;
 use std::ops::{Deref, DerefMut};
 use std::pin::Pin;
-use std::process;
-use std::sync::atomic::{AtomicUsize, Ordering};
 use std::task::{Context, Poll};
 
-use event_listener::{Event, EventListener};
+mod raw;
 
-use crate::futures::Lock;
-use crate::{Mutex, MutexGuard};
-
-const WRITER_BIT: usize = 1;
-const ONE_READER: usize = 2;
+use raw::*;
```

Comment:
I'd prefer to avoid a glob import here; just import the individual items from `raw`.